### PR TITLE
Neo4j driver metrics

### LINKED ISSUE #10188: Support exposing metrics from Neo4j driver
Neo4j driver has got built-in support for connection pool metrics. We just need to bridge them into our metrics extension to be able to automatically expose them to the outside world.

### PR DESCRIPTION
**Pull request by jmartisk**
Fixes #10188

### CODE REVIEW

**Comment by jmartisk** [comment]
@michael-simons

**Review by michael-simons — APPROVED** [review]
A big +1 from our side.  like the keen eye to the detail (when the pool is not being used). Well done.

File: docs/src/main/asciidoc/neo4j.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/neo4j.adoc:408]: I'm wondering if it would be better to have a global metrics setting rather than something under `pool`. Wouldn't it be more in line with the other extensions?

If we want to add other metrics, will we add more boolean settings?

**Review by gsmet — CHANGES_REQUESTED** [review]
Asked a small question about the config. Let's have this discussion before merging.

File: docs/src/main/asciidoc/neo4j.adoc
Comment (jmartisk) [reply docs/src/main/asciidoc/neo4j.adoc:408]: That's a very good point. At first I was planning to add a `quarkus.neo4j.metrics-enabled` property, but then I saw that there already exists `quarkus.neo4j.pool.metrics-enabled` which enables collection of metrics (internally without exporting them). So what I did instead of adding a new property (which would be confusing due to a very similar name), I changed the meaning of the original property, and now, rather than just enable collection of metrics, it also enables exporting them by the Metrics extension.

We could potentially rename the original one to `quarkus.neo4j.metrics-enabled` and let it do both these things at the same time, but it would break any usage of the original property, so I decided to go with the backward compatible approach. 

In the Agroal extension, we do have two different properties
`quarkus.datasource.jdbc.metrics-enabled` and
`quarkus.datasource.DS_NAME.jdbc.enable-metrics` 
There it makes sense to me because here you can enable metrics for each named data source separately, but in Neo4j we only support one connection pool, so it doesn't seem to have much value to have two separate properties for it.

Comment (gsmet) [reply docs/src/main/asciidoc/neo4j.adoc:408]: I think I would make it a global metric, deprecate the old one (but still consider it if it's true). If you use the `@Deprecated` annotation, the old one won't be included in the config reference doc.

Comment (gsmet) [reply docs/src/main/asciidoc/neo4j.adoc:408]: But better wait for @michael-simons 's feedback.

Comment (michael-simons) [reply docs/src/main/asciidoc/neo4j.adoc:408]: When I read the PR I expected the meaning of the property to stay the same plus when SmallRye Metrics is available that our pool metrics are exported (Probably because that's the way we role with Spring Boot, which I actually like).

Apart from that we don't have any plans for metrics apart the pool itself in the driver for the time being (Thinking query metrics (which I would like btw)). But if we do this in the future, I would prefer dedicated a dedicated property for it.

Comment (gsmet) [reply docs/src/main/asciidoc/neo4j.adoc:408]: OK, I suppose we can keep it that way then and wait for user complaints if any.

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
CI is very unhappy though :).

**Comment by jmartisk** [comment]
The failures looked unrelated to the PR, I force pushed without any change (is there a better way to restart the CI?)

**Comment by gsmet** [comment]
Yes, you need to go to the Checks tab, then on the Quarkus CI tab and click Re-run.

**Review by gastaldi — APPROVED** [review]
LGTM

**Pull request merged**
